libnavigation-ui/build.gradle

```diff
@@ -84,6 +84,14 @@ dependencies {
   // Mapbox Map SDK Javadoc
   javadocDeps dependenciesList.mapboxMapSdk
 
+  // Network
+  implementation(dependenciesList.okhttp) {
+    force = true
+  }
+  implementation(dependenciesList.okhttpInterceptor) {
+    force = true
+  }
+
```

Comment:
Same as above

Reply:
there will be compile error without these lines:
```
ERROR: Unable to resolve dependency for ':libnavigation-ui@debug/compileClasspath': Could not resolve com.squareup.okhttp3:okhttp:3.12.0.
Show Details
Affected Modules: libnavigation-ui


ERROR: Unable to resolve dependency for ':libnavigation-ui@debug/compileClasspath': Could not resolve com.squareup.okhttp3:okhttp:3.14.4.
Show Details
Affected Modules: libnavigation-ui


ERROR: Unable to resolve dependency for ':libnavigation-ui@debug/compileClasspath': Could not resolve com.squareup.okhttp3:okhttp:3.12.7.
Show Details
Affected Modules: libnavigation-ui
```

Reply:
detailed error looks like this:
```
> Task :libnavigation-ui:dependencyInsight
com.squareup.okhttp3:okhttp:3.10.0 FAILED
   Failures:
      - Could not resolve com.squareup.okhttp3:okhttp:3.10.0.
          - Cannot find a version of 'com.squareup.okhttp3:okhttp' that satisfies the version constraints: 
               Dependency path 'com.mapbox.navigation:libnavigation-ui:1.0.0-SNAPSHOT' --> 'com.squareup.picasso:picasso:2.71828' --> 'com.squareup.okhttp3:okhttp:3.10.0'
               Constraint path 'com.mapbox.navigation:libnavigation-ui:1.0.0-SNAPSHOT' --> 'com.squareup.okhttp3:okhttp:{strictly 3.12.0}' because of the following reason: debugRuntimeClasspath uses version 3.12.0
               Dependency path 'com.mapbox.navigation:libnavigation-ui:1.0.0-SNAPSHOT' --> 'com.mapbox.mapboxsdk:mapbox-android-sdk:9.1.0' --> 'com.mapbox.mapboxsdk:mapbox-android-telemetry:5.0.0' --> 'com.squareup.okhttp3:okhttp:3.12.0'
               Dependency path 'com.mapbox.navigation:libnavigation-ui:1.0.0-SNAPSHOT' --> 'com.mapbox.mapboxsdk:mapbox-sdk-services:5.1.0' --> 'com.mapbox.mapboxsdk:mapbox-sdk-core:5.1.0' --> 'com.squareup.retrofit2:retrofit:2.7.1' --> 'com.squareup.okhttp3:okhttp:3.14.4'
               Dependency path 'com.mapbox.navigation:libnavigation-ui:1.0.0-SNAPSHOT' --> 'com.mapbox.mapboxsdk:mapbox-sdk-services:5.1.0' --> 'com.mapbox.mapboxsdk:mapbox-sdk-core:5.1.0' --> 'com.squareup.okhttp3:logging-interceptor:3.12.7' --> 'com.squareup.okhttp3:okhttp:3.12.7'

com.squareup.okhttp3:okhttp:3.10.0 FAILED
\--- com.squareup.picasso:picasso:2.71828
     \--- debugCompileClasspath

com.squareup.okhttp3:okhttp:3.12.0 FAILED
   Selection reasons:
      - By constraint : debugRuntimeClasspath uses version 3.12.0
   Failures:
      - Could not resolve com.squareup.okhttp3:okhttp:{strictly 3.12.0}. (already reported)

com.squareup.okhttp3:okhttp:{strictly 3.12.0} -> 3.12.0 FAILED
\--- debugCompileClasspath

com.squareup.okhttp3:okhttp:3.12.0 FAILED
\--- com.mapbox.mapboxsdk:mapbox-android-telemetry:5.0.0
     +--- debugCompileClasspath (requested com.mapbox.mapboxsdk:mapbox-android-telemetry:{strictly 5.0.0})
     \--- com.mapbox.mapboxsdk:mapbox-android-sdk:9.1.0
          +--- debugCompileClasspath
          \--- com.mapbox.mapboxsdk:mapbox-android-plugin-annotation-v9:0.8.0 (requested com.mapbox.mapboxsdk:mapbox-android-sdk:9.0.0)
               \--- debugCompileClasspath

com.squareup.okhttp3:okhttp:3.14.4 FAILED
   Failures:
      - Could not resolve com.squareup.okhttp3:okhttp:3.14.4. (already reported)

com.squareup.okhttp3:okhttp:3.14.4 FAILED
\--- com.squareup.retrofit2:retrofit:2.7.1
     +--- debugCompileClasspath (requested com.squareup.retrofit2:retrofit:{strictly 2.7.1})
     +--- com.mapbox.mapboxsdk:mapbox-sdk-core:5.1.0
     |    +--- debugCompileClasspath (requested com.mapbox.mapboxsdk:mapbox-sdk-core:{strictly 5.1.0})
     |    \--- com.mapbox.mapboxsdk:mapbox-sdk-services:5.1.0
     |         \--- debugCompileClasspath
     \--- com.squareup.retrofit2:converter-gson:2.7.1
          +--- debugCompileClasspath (requested com.squareup.retrofit2:converter-gson:{strictly 2.7.1})
          \--- com.mapbox.mapboxsdk:mapbox-sdk-core:5.1.0 (*)

(*) - dependencies omitted (listed previously)

Reply:
Force to use the target okhttp version fixes the above issue.

Reply:
@JunDai 
🆗 took a stab at this and as you mentioned problem is Retrofit version brought by `mapbox-java`

One thing that we could explore though is publishing a [bill of materials](https://docs.gradle.org/6.2/userguide/platforms.html#sub:bom_import) (BOM) for our SDKs. Depending on this from Gradle or Maven would keep all of our SDKs artifacts on the same version, even if they're declared via transitive dependencies. You can even omit versions when declaring other SDKs dependencies. I think that's the right long-term solution. Could you take a look at it when you have a chance? We can also cut a follow up ticket to track that work and merge here. What do you think?